code/game/atoms_init.dm

```diff
@@ -106,12 +117,7 @@
 
 	forceMove(null)
 
-	if(LAZYLEN(movement_handlers) && !ispath(movement_handlers[1]))
-		QDEL_NULL_LIST(movement_handlers)
-
-	if (bound_overlay)
-		QDEL_NULL(bound_overlay)
+	vis_locs = null //clears this atom out of all vis_contents
+	clear_vis_contents(src)
 
-	if(virtual_mob && !ispath(virtual_mob))
-		qdel(virtual_mob)
-		virtual_mob = null
+	. = ..() // called last so that events are unregistered before the parent call
```

Comment:
Confusing comment: events are unregistered _in_ the parent call, what does this mean?

One thing this change does is re-order when the destroyed event is raised vs. everything else in this proc. Now it is raised after being moved to nullspace and deleting everything in `src`, which makes it impossible to preempt being deleted while inside something via destroyed events. I don't know how much code relies on this, but I also wouldn't be surprised if this breaks things (e.g. power).

Reply:
I think(?) that they mean movement handlers rather than events. Not seeing anything else in Destroy() that relates. Unsure about the ordering question.

Reply:
No, it has to be moved to nullspace prior to the parent call so that `cleanup_events()` doesn't run unnecessarily, since it's very slow. Moving to nullspace unregisters several events that are registered on `Entered()`. IMO a different pre-destroy event would be better for what you're talking about; IIRC in some PR I left a comment with a TODO relating to exactly that.

Reply:
Two questions from the Discord convo are -
- Where is the human registering to the crypod's events?
- Why not just dump everything into nullspace or unregister the events manually before qdel?